lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -1401,6 +1402,188 @@ static void maybeDiagnoseClassWithoutInitializers(ClassDecl *classDecl) {
   diagnoseClassWithoutInitializers(classDecl);
 }
 
+static bool isModuleQualified(TypeRepr *repr, ModuleDecl *module) {
+  auto extIdent = dyn_cast_or_null<CompoundIdentTypeRepr>(repr);
+  if (!extIdent) {
+    return false;
+  }
+  
+  auto components = extIdent->getComponents();
+  if (components.size() == 1) {
+    return false;
+  }
+  
+  return components.front()->getNameRef().isSimpleName(module->getName());
```

Comment:
@beccadax We'll have to remember to update this when module selectors go through.

Reply:
Harlan, could you add a comment starting with `// FIXME(ModQual):` somewhere in this function so I don't miss it?

Reply:
Done, thanks! I'm excited to see that eventually get traction!